plugin-viewport/src/main/kotlin/com/mapbox/maps/plugin/viewport/experimental/state/FollowingViewportStateImpl.kt

```diff
@@ -0,0 +1,165 @@
+package com.mapbox.maps.plugin.viewport.experimental.state
+
+import android.animation.AnimatorSet
+import android.animation.ValueAnimator
+import androidx.core.animation.doOnEnd
+import com.mapbox.geojson.Point
+import com.mapbox.maps.CameraOptions
+import com.mapbox.maps.plugin.animation.Cancelable
+import com.mapbox.maps.plugin.animation.camera
+import com.mapbox.maps.plugin.delegates.MapDelegateProvider
+import com.mapbox.maps.plugin.locationcomponent.OnIndicatorBearingChangedListener
+import com.mapbox.maps.plugin.locationcomponent.OnIndicatorPositionChangedListener
+import com.mapbox.maps.plugin.locationcomponent.location
+import com.mapbox.maps.plugin.viewport.experimental.data.FollowingViewportStateOptions
+import com.mapbox.maps.plugin.viewport.experimental.transition.MapboxViewportTransitionFactory
+import java.util.concurrent.*
+
+/**
+ * The actual implementation of [FollowingViewportState] that follows user's location.
+ */
+class FollowingViewportStateImpl internal constructor(
+  mapDelegateProvider: MapDelegateProvider,
+  /**
+   * Describes the configuration options of the state.
+   */
+  override var options: FollowingViewportStateOptions
```

Comment:
On iOS, I recalculate the camera when options changes. Is that a behavior we want?

Reply:
The camera options is calculated continuously on Android, so after updating the FollowingViewportStateOptions, the next location update, camera options will be calculated using the updated options.

Reply:
What if the next location isn't delivered for a few seconds?

Reply:
Good point, I can trigger a new update when the option has changed